delira/data_loading/data_manager.py

```diff
@@ -340,38 +78,30 @@ class defining the sampling strategy
         self.transforms = transforms
 
         if data_loader_cls is None:
-            logger.info("No DataLoader Class specified. Using BaseDataLoader")
-            data_loader_cls = BaseDataLoader
+            logger.info("No dataloader Class specified. Using DataLoader")
+            data_loader_cls = DataLoader
         else:
             assert inspect.isclass(data_loader_cls), \
                 "data_loader_cls must be class not instance of class"
-            assert issubclass(data_loader_cls, SlimDataLoaderBase), \
-                "dater_loader_cls must be subclass of SlimDataLoaderBase"
+            assert issubclass(data_loader_cls, DataLoader), \
+                "dater_loader_cls must be subclass of DataLoader"
 
         self.data_loader_cls = data_loader_cls
 
-        if isinstance(data, AbstractDataset):
-            self.dataset = data
-        else:
-
-            if dataset_cls is None:
-                if from_disc:
-                    dataset_cls = BaseLazyDataset
-                else:
-                    dataset_cls = BaseCacheDataset
-
-                logger.info("No DataSet Class specified. Using %s instead" %
-                            dataset_cls.__name__)
-
+        if not isinstance(data, AbstractDataset):
+            if isinstance(data, dict):
+                data = DictDataset(data)
+            elif isinstance(data, Iterable):
+                data = IterableDataset(data)
             else:
-                assert issubclass(dataset_cls, AbstractDataset), \
-                    "dataset_cls must be subclass of AbstractDataset"
+                raise TypeError("Invalid Data type given: %s"
+                                % type(data).__name__)
```

Comment:
Duplicate to `DataLoader`. I would prefer adding a function for this or it should be handled at one place.
(And a test is missing for this ^^)

Reply:
I removed this here in favor of dynamically creating the sampler after the dataloader